crates/core/src/config.rs

```diff
@@ -0,0 +1,178 @@
+const DEFAULT_IP_V4_ADDR: &str = "127.0.0.1";
+const DEFAULT_PORT: u16 = 3030;
+
+// mainnet constants
+pub const MAINNET_CC_ADDRESS: &str = "c662c410C0ECf747543f5bA90660f6ABeBD9C8c4";
+pub const MAINNET_CONSENSUS_RPC: &str = "https://www.lightclientdata.org";
+pub const MAINNET_FALLBACK_RPC: &str = "https://sync-mainnet.beaconcha.in";
+
+// testnet constants
+pub const TESTNET_CC_ADDRESS: &str = "de29d060D45901Fb19ED6C6e959EB22d8626708e";
+pub const TESTNET_CONSENSUS_RPC: &str = "http://testing.prater.beacon-api.nimbus.team";
+pub const TESTNET_FALLBACK_RPC: &str = "https://sync-goerli.beaconcha.in";
+
+/// global config
+#[derive(Clone, Deserialize, Debug)]
+pub struct Config {
+    pub network: Network,
+    pub eth_execution_rpc: String,
+    pub starknet_rpc: String,
+    #[serde(default = "data_dir")]
+    pub data_dir: PathBuf,
+    #[serde(default = "poll_secs")]
+    pub poll_secs: u64,
+    #[serde(default = "rpc_addr")]
+    pub rpc_addr: SocketAddr,
+    #[serde(default = "fee_token_addr")]
+    pub fee_token_addr: FieldElement,
+}
+
+fn data_dir() -> PathBuf {
+    PathBuf::from(DEFAULT_DATA_DIR)
+}
+
+fn poll_secs() -> u64 {
+    DEFAULT_POLL_SECS
+}
+
+fn rpc_addr() -> SocketAddr {
+    SocketAddr::from_str(&format!("{DEFAULT_IP_V4_ADDR}:{DEFAULT_PORT}")).unwrap()
+}
+
+fn fee_token_addr() -> FieldElement {
+    FieldElement::from_hex_be(DEFAULT_FEE_TOKEN_ADDR).unwrap()
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            network: Network::MAINNET,
+            eth_execution_rpc: "http://localhost:5054".to_string(),
+            starknet_rpc: "http://localhost:9545".to_string(),
+            data_dir: data_dir(),
+            poll_secs: poll_secs(),
+            rpc_addr: rpc_addr(),
+            fee_token_addr: fee_token_addr(),
+        }
+    }
+}
+
+impl Config {
+    pub fn from_env() -> Self {
+        Self {
+            network: Network::from_str(&std::env::var("NETWORK").unwrap_or_default()).unwrap_or(Network::MAINNET),
+            eth_execution_rpc: std::env::var("ETH_EXECUTION_RPC").unwrap_or_default(),
+            starknet_rpc: std::env::var("STARKNET_RPC").unwrap_or_default(),
+            data_dir: PathBuf::from(std::env::var("DATA_DIR").unwrap_or_default()),
+            poll_secs: u64::from_str(&std::env::var("POLL_SECS").unwrap_or_default()).unwrap_or(DEFAULT_POLL_SECS),
+            rpc_addr: SocketAddr::from_str(&format!("{DEFAULT_IP_V4_ADDR}:{DEFAULT_PORT}")).unwrap(),
+            fee_token_addr: fee_token_addr(),
+        }
+    }
+
+    pub fn from_file(path: &str) -> Self {
+        let raw_conf = fs::read_to_string(path).unwrap_or_else(|_| panic!("unable to read file: {path}"));
+
+        if path.contains(".toml") {
+            toml::from_str(&raw_conf).unwrap()
+        } else if path.contains(".json") {
+            serde_json::from_str(&raw_conf).unwrap()
+        } else {
+            Self::default()
```

Comment:
a bit tricky

Reply:
tricky i.e. shouldn't set default if there isn't any config file?

Am good w/ panicing if there isn't a good config where the user pointed

Reply:
Yeah. Better panic because it’s explicit. 
As a bad idea, it can be simply `--conf` `path_to_file` or `default` or `env`

Reply:
I like the idea to panic here. That way if they have old env vars set and init w/ a conf path they don't end up with unexpected state.